### ISSUE TITLE
Refactor out TPE for Flow stuff

### ISSUE THREAD

**Issue by pzakatov** [telstra/open-kilda#1128]
Blocked by: 
- #1127 Create java based DAO layer in front of Neo 
- #561 Eliminate Cache topology

**Comment by timofei-durakov** [comment]
### TE
#1128
- [ ] Design documentation **Est. 2d**
- [ ] Reimplement CRUD Flow operations in Storm **Est. 3-4d**
  - Implement handler for flow requests from NB to WFM
  - Move flow operations part to flow topology
  - NB related operations are out of scope

- [ ] Flow install and delete operations **Est. 5-6d** 
  - implementation of transit VLAN's per switch are **out of scope**

- [ ] Reimplement Switch operations in Storm **Est. 2d**
  - Create SwitchService
  - Create NetworkTopologyBolt in Event Topology
  - Get rid of TPE bolts

- [ ] Reimplement Isl operations in Storm **Est. 3d**
  - Create IslService
  - Create NetworkTopologyBolt in Event Topology
  - Get rid of TPE bolts

- [ ] Reimplement Port operations in Storm **Est. 2d**
  - Create PortService
  - Create NetworkTopologyBolt in Event Topology
  - Get rid of TPE bolts

- [ ] Reimplement Features in separate topology or Northbound. **Out Of Scope**
  - Create FeaturesBolt
  - Implement Feature Service


- [ ] Reimplement Validate Rules pipeline in NBtopology **Out Of Scope**
  - Create RulesValidationBolt
  - Add it to nb worker topology

- [ ] Reimplement Link props operations in WFM **Est. 3-4d**
  - Create a bolt for handling such request
  - Add it to nb worker topology

- [ ] Reimplement sync rules operation in wfm **Out Of Scope**
  - Create a bolt in nbworker topology to handle sync requests

**Comment by timofei-durakov** [comment]
design is done in https://github.com/telstra/open-kilda/pull/1402
